src/main/java/com/kaziamyr/onlinebookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package com.kaziamyr.onlinebookstore.service.impl;
+
+import com.kaziamyr.onlinebookstore.dto.cartitem.CartItemDto;
+import com.kaziamyr.onlinebookstore.dto.cartitem.CreateCartItemRequestDto;
+import com.kaziamyr.onlinebookstore.dto.cartitem.PutCartItemRequestDto;
+import com.kaziamyr.onlinebookstore.dto.shoppingcart.ShoppingCartDto;
+import com.kaziamyr.onlinebookstore.exception.EntityNotFoundException;
+import com.kaziamyr.onlinebookstore.mapper.CartItemMapper;
+import com.kaziamyr.onlinebookstore.mapper.ShoppingCartMapper;
+import com.kaziamyr.onlinebookstore.model.CartItem;
+import com.kaziamyr.onlinebookstore.model.ShoppingCart;
+import com.kaziamyr.onlinebookstore.model.User;
+import com.kaziamyr.onlinebookstore.repository.BookRepository;
+import com.kaziamyr.onlinebookstore.repository.CartItemRepository;
+import com.kaziamyr.onlinebookstore.repository.ShoppingCartRepository;
+import com.kaziamyr.onlinebookstore.service.ShoppingCartService;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public ShoppingCartDto getByUser() {
+        ShoppingCart shoppingCart = getOrCreateUsersShoppingCart();
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public CartItemDto addBookToShoppingCart(CreateCartItemRequestDto request) {
+        ShoppingCart shoppingCart = getOrCreateUsersShoppingCart();
+        List<CartItem> cartItems = new ArrayList<>(shoppingCart.getCartItems());
+        CartItem cartItem;
+        Optional<CartItem> optionalPresentCartItem = cartItems.stream()
+                .filter(cartItem1 -> cartItem1.getBook().getId().equals(request.getBookId()))
+                .findAny();
+        if (optionalPresentCartItem.isPresent()) {
+            cartItem = optionalPresentCartItem.get();
+            cartItem.setQuantity(cartItem.getQuantity() + request.getQuantity());
+            cartItems.set(cartItems.indexOf(optionalPresentCartItem.get()), cartItem);
+        } else {
+            cartItem = cartItemMapper.toModel(request);
```

Comment:
Move body of this else to method and call it something like addNewBook. It will significantly improve readability of code and please do same for other similar cases